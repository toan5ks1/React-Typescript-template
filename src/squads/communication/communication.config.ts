import { commonBuildPipelineConfig } from "../../../mfe/common.config";

export default commonBuildPipelineConfig("communication-main", {
    entry: "./exports/applications/communication-main.tsx",
    port: 4001,
    rootDir: __dirname,
    checkConfig: {
        typescript: true,
        eslint: {
            files: ["./src"],
            extensions: [".ts", ".tsx"],
        },
    },
});
