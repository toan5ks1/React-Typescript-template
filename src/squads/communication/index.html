<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <meta name="theme-color" content="#000000" />
        <meta name="description" content="Communication micro" />
        <title>Communication micro</title>
    </head>
    <body>
        <noscript>You need to enable JavaScript to run this app.</noscript>
        <div id="root"></div>
        <script>
            window.global = globalThis;
        </script>
        <script type="module" src="./exports/applications/communication-main.tsx"></script>
    </body>
</html>
