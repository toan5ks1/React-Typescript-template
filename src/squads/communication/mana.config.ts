export default {
    application: {
        teamName: "communication",
        sources: {
            "@manabie-com/mfe/communication-main": {
                outDir: "./build/communication-main",
                configPath: "./communication.config.ts",
                entryPath: "communication-main/index.js",
            },
        },
        hooks: {
            beforeBuild: async () => {},
        },
    },
};
