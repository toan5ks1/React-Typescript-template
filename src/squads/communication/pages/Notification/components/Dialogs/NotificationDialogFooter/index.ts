export { default } from "./NotificationDialogFooter";
export * from "./NotificationDialogFooter";
