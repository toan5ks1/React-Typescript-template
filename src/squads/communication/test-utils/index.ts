export { default as TestThemeProvider } from "./TestThemeProvider";
export { default as TestApp } from "./TestApp";
