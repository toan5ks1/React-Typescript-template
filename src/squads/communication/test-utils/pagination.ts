import { PaginationWithTotal } from "@manabie-com/react-utils";

export const createMockPaginationWithTotalObject = (
    totalItems: number = 10,
    currentPage: number = 0 // Because first page have value equal 0
): PaginationWithTotal => {
    return {
        count: totalItems,
        limit: totalItems,
        page: currentPage,
        rowsPerPage: totalItems,
        onPageChange: jest.fn(),
        onRowsPerPageChange: jest.fn(),
        offset: totalItems,
    };
};
