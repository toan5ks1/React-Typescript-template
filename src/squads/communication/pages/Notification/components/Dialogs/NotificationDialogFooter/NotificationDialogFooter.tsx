import { Box, Grid } from "@mui/material";
import ButtonDelete from "src/components/Buttons/ButtonDelete";
import { DialogHFProps } from "src/components/Dialogs/types";
import DialogNotificationButtonGroup from "src/squads/communication/pages/Notification/components/DialogNotificationButtonGroup";

import useTranslate from "src/squads/communication/hooks/useTranslate";

export interface NotificationDialogFooterProps<T> extends Pick<DialogHFProps<T>, "onSave"> {
    onSaveDraft: () => void;
    onDiscard: () => void;
    onSaveSchedule: () => void;
    isSendingNotification?: boolean;
    isCreatingNotification?: boolean;
    isSavingScheduleNotification?: boolean;
    shouldEnableScheduleMode?: boolean;
}

const NotificationDialogFooter = <T extends any>(props: NotificationDialogFooterProps<T>) => {
    const {
        onDiscard,
        onSave,
        onSaveDraft,
        onSaveSchedule,
        isCreatingNotification,
        isSendingNotification,
        isSavingScheduleNotification,
        shouldEnableScheduleMode,
    } = props;
    const t = useTranslate();

    const shouldFooterDisabled = isSendingNotification || isCreatingNotification;

    return (
        <Grid container data-testid="NotificationDialogFooter__actions">
            <Grid item xs={6}>
                <ButtonDelete
                    startIcon={undefined}
                    disabled={shouldFooterDisabled}
                    onClick={onDiscard}
                    data-testid="NotificationDialogFooter__buttonDiscard"
                >
                    {t("ra.common.action.discard")}
                </ButtonDelete>
            </Grid>

            <Grid item xs={6}>
                <Box display="flex" justifyContent="flex-end">
                    <DialogNotificationButtonGroup
                        onSave={onSave!}
                        onSaveDraft={onSaveDraft}
                        onSaveSchedule={onSaveSchedule}
                        isSavingScheduleNotification={isSavingScheduleNotification!}
                        isCreatingNotification={isCreatingNotification!}
                        isSendingNotification={isSendingNotification!}
                        shouldFooterDisabled={shouldFooterDisabled!}
                        shouldEnableScheduleMode={shouldEnableScheduleMode!}
                    />
                </Box>
            </Grid>
        </Grid>
    );
};

export default NotificationDialogFooter;
