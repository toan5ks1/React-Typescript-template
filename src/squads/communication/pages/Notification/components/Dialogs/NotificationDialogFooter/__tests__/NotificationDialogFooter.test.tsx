import { FieldValues } from "react-hook-form";
import { TestThemeProvider } from "src/squads/communication/test-utils";

import NotificationDialogFooter, {
    NotificationDialogFooterProps,
} from "../NotificationDialogFooter";

import { render, RenderResult, screen } from "@testing-library/react";
import TestHookFormProvider from "src/squads/communication/test-utils/TestHookFormProvider";

const props: NotificationDialogFooterProps<FieldValues> = {
    onSave: jest.fn(),
    onSaveDraft: jest.fn(),
    onDiscard: jest.fn(),
    onSaveSchedule: jest.fn(),
};

describe("<NotificationDialogFooter/>", () => {
    let wrapper: RenderResult;

    beforeEach(() => {
        wrapper = render(
            <TestThemeProvider>
                <TestHookFormProvider>
                    <NotificationDialogFooter {...props} />
                </TestHookFormProvider>
            </TestThemeProvider>
        );
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should render correct UI", () => {
        expect(
            screen.queryByTestId("DialogNotificationButtonGroup__buttonSaveDraft")
        ).toBeInTheDocument();
        expect(
            screen.queryByTestId("DialogNotificationButtonGroup__buttonSend")
        ).toBeInTheDocument();
        expect(screen.queryByTestId("NotificationDialogFooter__buttonDiscard")).toBeInTheDocument();
    });
});
