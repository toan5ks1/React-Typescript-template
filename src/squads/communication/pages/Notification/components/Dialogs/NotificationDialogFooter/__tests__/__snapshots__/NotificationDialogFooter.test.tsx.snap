// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationDialogFooter/> should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 50%;
}

@media (min-width:600px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:960px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:1440px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:1920px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 0.875rem;
  font-weight: 500;
  letter-spacing: 0.4px;
  text-transform: capitalize;
  font-family: Roboto,sans-serif;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #E0E0E0;
  border-radius: 4px;
  color: #212121;
  color: #F44336;
}

.emotion-2::-moz-focus-inner {
  border-style: none;
}

.emotion-2.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-2 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(224, 224, 224, 0.04);
}

@media (hover: none) {
  .emotion-2:hover {
    background-color: transparent;
  }
}

.emotion-2.Mui-disabled {
  color: #BDBDBD;
}

.emotion-2:hover {
  background-color: rgba(33, 33, 33, 0.04);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-5 {
  margin-right: 16px;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 0.875rem;
  font-weight: 500;
  letter-spacing: 0.4px;
  text-transform: capitalize;
  font-family: Roboto,sans-serif;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #E0E0E0;
  border-radius: 4px;
  color: #212121;
}

.emotion-6::-moz-focus-inner {
  border-style: none;
}

.emotion-6.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-6 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-6:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(224, 224, 224, 0.04);
}

@media (hover: none) {
  .emotion-6:hover {
    background-color: transparent;
  }
}

.emotion-6.Mui-disabled {
  color: #BDBDBD;
}

.emotion-6:hover {
  background-color: rgba(33, 33, 33, 0.04);
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 0.875rem;
  font-weight: 500;
  letter-spacing: 0.4px;
  text-transform: capitalize;
  font-family: Roboto,sans-serif;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #2196F3;
  box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2),0px 2px 2px rgba(0, 0, 0, 0.14),0px 1px 5px rgba(0, 0, 0, 0.12);
  border-radius: 4px;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #0B79D0;
  box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px rgba(0, 0, 0, 0.14),0px 1px 10px rgba(0, 0, 0, 0.12);
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: #2196F3;
  }
}

.emotion-7:active {
  box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12);
}

.emotion-7.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px rgba(0, 0, 0, 0.14),0px 1px 18px rgba(0, 0, 0, 0.12);
}

.emotion-7.Mui-disabled {
  color: #BDBDBD;
  box-shadow: none;
  background-color: #E0E0E0;
}

<div>
  <form
    novalidate=""
  >
    <div
      class="MuiGrid-root MuiGrid-container emotion-0"
      data-testid="NotificationDialogFooter__actions"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-1"
      >
        <button
          aria-label="ra.common.action.delete"
          class="MuiButton-root MuiButton-text MuiButton-textDefault MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  emotion-2"
          data-testid="NotificationDialogFooter__buttonDiscard"
          style="word-break: keep-all;"
          tabindex="0"
          type="button"
        >
          ra.common.action.discard
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-1"
      >
        <div
          class="MuiBox-root emotion-4"
        >
          <div
            class="MuiBox-root emotion-5"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textDefault MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  emotion-6"
              data-testid="DialogNotificationButtonGroup__buttonSaveDraft"
              tabindex="0"
              type="button"
            >
              ra.common.saveDraft
            </button>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  emotion-7"
            data-testid="DialogNotificationButtonGroup__buttonSend"
            style="word-break: keep-all;"
            tabindex="0"
            type="button"
          >
            ra.common.send
          </button>
        </div>
      </div>
    </div>
    <input
      data-testid="TextFieldHF__submit"
      type="submit"
    />
  </form>
</div>
`;
